frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,202 @@ import {
 import React from "react";
 
 import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import UserAPIClient from "../../../APIClients/UserAPIClient";
+import VolunteerAPIClient from "../../../APIClients/VolunteerAPIClient";
+import { Role, Status } from "../../../types/AuthTypes";
 import { DonorResponse } from "../../../types/DonorTypes";
+import { VolunteerResponse } from "../../../types/VolunteerTypes";
+import { UserMgmtTableRecord } from "./types";
 
 const UserManagementPage = (): JSX.Element => {
   const [donors, setDonors] = React.useState([] as DonorResponse[]);
+  const [volunteers, setVolunteers] = React.useState([] as VolunteerResponse[]);
+  const [users, setUsers] = React.useState([] as UserMgmtTableRecord[]); // both donors and volunteers
   const [search, setSearch] = React.useState("");
+  const [selectedFilter, setSelectedFilter] = React.useState("all");
+  const [dataChanged, setDataChanged] = React.useState<0 | 1>(0);
 
-  const tableData = React.useMemo(() => {
-    if (!search) return donors;
-    return donors.filter(
-      (donor) =>
-        donor.businessName.toLowerCase().includes(search.toLowerCase()) ||
-        donor.email.toLowerCase().includes(search.toLowerCase()) ||
-        donor.firstName
-          .toLowerCase()
-          .concat(" ", donor.lastName.toLowerCase())
-          .includes(search.toLowerCase()) ||
-        donor.phoneNumber.toLowerCase().includes(search.toLowerCase()),
-    );
-  }, [search, donors]);
+  const filterOptions = [
+    {
+      value: "all",
+      label: "All accounts",
+    },
+    {
+      value: "volunteers",
+      label: "Volunteers",
+    },
+    {
+      value: "donors",
+      label: "Donors",
+    },
+  ];
+
+  // Merge donors and volunteers into one list of type UserMgmtTableRecord
+  const buildTableUsers = (): UserMgmtTableRecord[] => {
+    const mergedUsers: UserMgmtTableRecord[] = [];
+
+    volunteers.forEach(function (volunteer) {
+      mergedUsers.push({
+        userId: volunteer.userId,
+        id: volunteer.id,
+        pointOfContact: `${volunteer.firstName} ${volunteer.lastName}`,
+        company: "",
+        email: volunteer.email,
+        phoneNumber: volunteer.phoneNumber,
+        accountType: volunteer.role,
+        approvalStatus: volunteer.status,
+      });
+    });
+
```

Comment:
do you know if the volunteers that are pending should be taking precedence in the table?

Reply:
@hanlinc27 getting confirmation on this 

(originally we wanted sorted by date created, and approval status doesn't matter (so no to pending taking precedence), but currently i have that volunteers are on top since the donors are appended to this list

Reply:
Anya says this is fine